File: app/src/main/java/com/myjb/dev/network/PriceInquiry2Yes24.java
package com.myjb.dev.network;

import android.support.annotation.NonNull;

import com.myjb.dev.model.ServiceModel;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

public class PriceInquiry2Yes24 extends PriceInquiry {

    protected final static String TAG = "PriceInquiry2Yes24";

    public PriceInquiry2Yes24(@NonNull String query, @NonNull OnPriceListener listener) {
        super(query, listener);
        baseUrl = "http://www.yes24.com/Mall/buyback/Search?CategoryNumber=018&SearchWord=";
    }

    @NonNull
    @Override
    protected Elements getElements(Document doc) {
        Elements elements = super.getElements(doc);

        Element element = elements.first();

        if (element == null)
            return new Elements();

        Elements children = element.children();

        if (children == null)
            return new Elements();

        return children;
    }

    @Override
    protected int getCompany() {
        return ServiceModel.Company.YES24;
    }

    @NonNull
    @Override
    protected String getBasicFilter() {
        return "ul[class=clearfix]";
    }

    @NonNull
    @Override
    protected String getISBN(Element element) {
        return element.select("span[class=bbG_isbn13]").text().split("-")[1];
    }

    @NonNull
    @Override
    protected String getNameFilter() {
        return "strong[class=name]";
    }

    @NonNull
    @Override
    protected String getPriceFilter() {
        return "td";
    }
}

File: app/src/main/java/com/myjb/dev/network/PriceInquiry2Yes24.kt
package com.myjb.dev.network

import com.myjb.dev.model.ServiceModel
import org.jsoup.nodes.Document
import org.jsoup.nodes.Element
import org.jsoup.select.Elements

class PriceInquiry2Yes24(query: String, listener: OnPriceListener) : PriceInquiry(query, listener) {
    init {
        baseUrl = "http://www.yes24.com/Mall/buyback/Search?CategoryNumber=018&SearchWord="
    }

    override fun getElements(doc: Document?): Elements {
        val elements = super.getElements(doc)

        val element = elements.first() ?: return Elements()

        val children = element.children() ?: return Elements()

        return children
    }

    override val company: Int
        get() = ServiceModel.Company.YES24

    override val basicFilter: String
        get() = "ul[class=clearfix]"

    override fun getISBN(element: Element): String {
        return element.select("span[class=bbG_isbn13]").text().split("-".toRegex())
            .dropLastWhile { it.isEmpty() }.toTypedArray()[1]
    }

    override val nameFilter: String
        get() = "strong[class=name]"

    override val priceFilter: String
        get() = "td"

    companion object {
        protected const val TAG: String = "PriceInquiry2Yes24"
    }
}
